### ISSUE TITLE
Make delete more productive?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#6016]
* Should we permit delete to take a comma-separated list of things to delete?

````chapel
delete obj1, obj2, obj3;
````

* If we have an array of objects, should we permit delete to promote?

````chapel
delete ArrOfObj;
````

rather than:

````chapel
forall a in ArrOfObj do delete a;
````

**Comment by vasslitvinov** [comment]
I say "yes" to both.

For the array case, did you mean `forall` instead of `for` ?

**Comment by bradcray** [comment]
> For the array case, did you mean forall instead of for ?

I was mostly trying to capture the need to do an explicit loop, but since I'm proposing we use a forall in the implementation, I've switched it now.

**Comment by psahabu** [comment]
@bradcray should we close this issue given that we've merged the branch?

**Comment by bradcray** [comment]
Oops, yes.  Addressed in #7424
